frontends/pytorch/csrc/c10_dispatch/acap_dispatch.h

```diff
@@ -0,0 +1,60 @@
+//===- acap_dispatch.h ------------------------------------------*- C++ -*-===//
+//
+// This file is licensed under a pytorch-style license
+// See frontends/pytorch/LICENSE for license information.
+//
+//===----------------------------------------------------------------------===//
+// "ATen Capture" dispatcher: Defines facility for capturing programs by
+// registering dispatch keys to intercept op execution.
+// References:
+//   http://blog.ezyang.com/2020/09/lets-talk-about-the-pytorch-dispatcher/
```

Comment:
Wow, that's very good documentation. And the design is super clean too!

Reply:
I know, right? I think they are still iterating on namespaces and such but the design is really nice. As a concrete number, doing this kind of interception the old way was about ~32K lines of code and was very fragile and not open to extension. In the new way, it is 308 (lines, not kilo-lines). Not completely done yet, so that will grow a bit... but still. It took me ~an afternoon to grok the entire codebase around these concepts and integrate with it.